### ISSUE TITLE
HelmRelease doesn't continually re-reconcile on Argo Workflow Step

### ISSUE THREAD

**Issue by jonathan-innis** [Azure/orkestra#213]
**Describe the bug**
In cases where the interval is long and the `HelmRelease` doesn't reconcile immediately, it can take up to the `interval` specified for the helm chart for the `HelmRelease` to re-reconcile. In some cases where this time > 5m, this can fail the entire workflow.

**To Reproduce**
1. Install the Orkestra chart
2. Install the `./config/samples/bookinfo.yaml` (this issue is intermittent so you may not see this on the first go-around)

**Expected behavior**
To avoid ignoring user-specified reconcile interval, we should install the helm-release and force re-reconcile on a polling interval using the reconcile annotation provided through `HelmRelease` api

**Additional context**
More information on how to force re-reconcile can be seen within the `flux reconcile` portion of the Flux CLI
https://github.com/fluxcd/flux2/blob/main/cmd/flux/reconcile.go

**Comment by nitishm** [comment]
So this is a delay caused by the helm-controller reconciliation interval? If so can we always keep it short since we don't necessarily need it to be tweaked by the end user?

**Comment by jonathan-innis** [comment]
Yes, it is. I think that we should re-reconcile very quickly while we are checking the status with the argo workflow, but not during standard reconciliation, this is going to put a lot of weight on the api server for no reason

**Comment by jonathan-innis** [comment]
@nitishm  I'm picking this up and prioritizing this above testing because otherwise our test cases are going to take forever

**Comment by jonathan-innis** [comment]
As I'm looking at it, it seems to be that we need to force re-reconcile after updating the HelmRepository/pushing the helm charts, I'm gonna try this, but I suspect this will fix it 😄
